src/swarm-engine/Swarm/Game/Step.hs

```diff
@@ -338,7 +339,11 @@ hypotheticalWinCheck em g ws oc = do
     Unwinnable _ -> grantAchievement LoseScenario
     _ -> return ()
 
-  messageInfo . announcementQueue %= (>< Seq.fromList (map ObjectiveCompleted $ completionAnnouncementQueue finalAccumulator))
+  queue <- messageInfo . announcementQueue Swarm.Util.<%= (>< Seq.fromList (map ObjectiveCompleted $ completionAnnouncementQueue finalAccumulator))
+  shouldPause <- use $ temporal . pauseOnCompletion
+
+  when (newWinState /= Ongoing || (notNull queue && shouldPause == PauseOnAnyObjective)) $
```

Comment:
I'm trying to make sure I understand the logic here.  What is the purpose of the `notNull queue` check?  How does the `newWinState` check and the other check fit together?  (And can you turn your answer into some comments in the code here?)

Reply:
I named the parts using `let`. Is it clearer now? 🙂

Reply:
Yes, much better, thanks!